apps/explorer/src/app/routes/parties/id/components/party-block-deposits.tsx

```diff
@@ -0,0 +1,164 @@
+import { t } from '@vegaprotocol/i18n';
+import { useNavigate } from 'react-router-dom';
+import { Routes } from '../../../../routes/route-names';
+import { AsyncRenderer, Button } from '@vegaprotocol/ui-toolkit';
+import { PartyBlock } from './party-block';
+import type { AccountFields } from '@vegaprotocol/accounts';
+import { useExplorerPartyDepositsWithdrawalsQuery } from './__generated__/Party-deposits-withdrawals';
+import {
+  type DepositOrWithdrawal,
+  combineDepositsWithdrawals,
+} from './lib/combine-deposits-withdrawals';
+import { Table, TableCell, TableRow } from '../../../../components/table';
+import {
+  EthExplorerLinkTypes,
+  ExternalExplorerLink,
+} from '../../../../components/links/external-explorer-link/external-explorer-link';
+import AssetBalance from '../../../..//components/asset-balance/asset-balance';
+import { Time } from '../../../../components/time';
+import {
+  DepositWithdrawalStatusIcon,
+  getDepositWithdrawalStatusLabel,
+} from './party-deposits-withdrawals-status-icon';
+import { useScreenDimensions } from '@vegaprotocol/react-helpers';
+import { useMemo } from 'react';
+
+export interface PartyBlockAccountProps {
+  partyId: string;
+  accountData: AccountFields[] | null;
+  accountLoading: boolean;
+  accountError?: Error;
+}
+
+/**
+ * Displays deposits & withdrawals for a party, sorted most-recent-first, with a button to show all.
+ * By default it is limited to 5 entries on the party by ID page, then in a nested route it can show
+ * all of them in
+ */
+export const PartyBlockDeposits = ({
+  partyId,
+  accountLoading,
+  accountError,
+}: PartyBlockAccountProps) => {
+  const navigate = useNavigate();
+
+  const { data, error, loading } = useExplorerPartyDepositsWithdrawalsQuery({
+    variables: { partyId },
+  });
+
+  const sortedData = data ? combineDepositsWithdrawals(data) : [];
+
+  const { screenSize } = useScreenDimensions();
+  const isTruncated = useMemo(
```

Comment:
Only use useMemo if you actually having something intensive enough. Usually this is worse for perf

Theres a few other spots in this PR where I'd remove useMemo too

Reply:
Removing them 👍